fails to build with `>=glibc-2.43`

Anologous to https://github.com/flatpak/flatpak/pull/6490

```bash
[2/4] clang -Ixdg-dbus-proxy.p -I. -I../xdg-dbus-proxy-0.1.6 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/lib64/libffi/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -D_GNU_SOURCE -Werror=aggregate-return -Werror=empty-body -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=init-self -Werror=int-conversion -Werror=misleading-indentation -Werror=missing-declarations -Werror=missing-include-dirs -Werror=missing-prototypes -Werror=overflow -Werror=pointer-arith -Werror=return-type -Werror=shadow -Werror=strict-prototypes -Werror=switch-default -Wswitch-enum -Wno-missing-field-initializers -Wno-error=missing-field-initializers -Wno-unused-parameter -Wno-error=unused-parameter -O2 -pipe -pthread -MD -MQ xdg-dbus-proxy.p/dbus-proxy.c.o -MF xdg-dbus-proxy.p/dbus-proxy.c.o.d -o xdg-dbus-proxy.p/dbus-proxy.c.o -c ../xdg-dbus-proxy-0.1.6/dbus-proxy.c
FAILED: [code=1] xdg-dbus-proxy.p/dbus-proxy.c.o 
clang -Ixdg-dbus-proxy.p -I. -I../xdg-dbus-proxy-0.1.6 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/lib64/libffi/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -D_GNU_SOURCE -Werror=aggregate-return -Werror=empty-body -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=init-self -Werror=int-conversion -Werror=misleading-indentation -Werror=missing-declarations -Werror=missing-include-dirs -Werror=missing-prototypes -Werror=overflow -Werror=pointer-arith -Werror=return-type -Werror=shadow -Werror=strict-prototypes -Werror=switch-default -Wswitch-enum -Wno-missing-field-initializers -Wno-error=missing-field-initializers -Wno-unused-parameter -Wno-error=unused-parameter -O2 -pipe -pthread -MD -MQ xdg-dbus-proxy.p/dbus-proxy.c.o -MF xdg-dbus-proxy.p/dbus-proxy.c.o.d -o xdg-dbus-proxy.p/dbus-proxy.c.o -c ../xdg-dbus-proxy-0.1.6/dbus-proxy.c
../xdg-dbus-proxy-0.1.6/dbus-proxy.c:141:5: error: assigning to 'guchar *' (aka 'unsigned char *') from 'const void *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
  141 |   s = memchr (remainder, 0, remainder_len);
      |     ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../xdg-dbus-proxy-0.1.6/dbus-proxy.c:149:9: error: assigning to 'guchar *' (aka 'unsigned char *') from 'const void *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
  149 |       s = memchr (remainder, 0, remainder_len);
      |         ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
[3/4] clang -Ixdg-dbus-proxy.p -I. -I../xdg-dbus-proxy-0.1.6 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/lib64/libffi/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -D_GNU_SOURCE -Werror=aggregate-return -Werror=empty-body -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=init-self -Werror=int-conversion -Werror=misleading-indentation -Werror=missing-declarations -Werror=missing-include-dirs -Werror=missing-prototypes -Werror=overflow -Werror=pointer-arith -Werror=return-type -Werror=shadow -Werror=strict-prototypes -Werror=switch-default -Wswitch-enum -Wno-missing-field-initializers -Wno-error=missing-field-initializers -Wno-unused-parameter -Wno-error=unused-parameter -O2 -pipe -pthread -MD -MQ xdg-dbus-proxy.p/flatpak-proxy.c.o -MF xdg-dbus-proxy.p/flatpak-proxy.c.o.d -o xdg-dbus-proxy.p/flatpak-proxy.c.o -c ../xdg-dbus-proxy-0.1.6/flatpak-proxy.c
ninja: build stopped: subcommand failed.
```

--- a/dbus-proxy.c
+++ b/dbus-proxy.c
@@ -133,7 +133,7 @@ add_args (GBytes    *bytes,
 {
   gsize data_len, remainder_len;
   const guchar *data = g_bytes_get_data (bytes, &data_len);
-  guchar *s;
+  const guchar *s;
   const guchar *remainder;
 
   remainder = data;

